### ISSUE TITLE
Home expiry warning should be relative to the configured timelock

### ISSUE THREAD

**Issue by darosior** [wizardsardine/liana#461]
It seems that #455 made it so that the warning will be displayed if there is less than a day worth of blocks before a recovery path is available. (@edouardparis please correct me if i'm wrong.)

But this arbitrary threshold doesn't generalize well. For instance if one set their recovery path to 1 year, we should definitely warn them sooner than a day before their coin(s) expire(s).

That's why i suggested using 10% of the configured timelock value in #439. But it can also be a smaller percentage. 1%? 5%?

5% generalizes fine: for a timelock of 1year it'll display the warning 2 weeks and a half before it becomes available. For a timelock of 3 months it'll be a bit less than 5 days before. For a maximum timelock of a year and (around) 3 months, it'll warn a bit more than 3 weeks before the recovery path becomes available.

What do you think @edouardparis @kloaec @mariabmont?

**Comment by kloaec** [comment]
I would strongly suggest to make this configurable in the settings. If we need a default option, probably 20%+, as it's playing with fire if you already lapsed 80% imho.

A frequent user (weekly) might want a ~2 week notice minimum, even for a 2 months timelock. An infrequent user (monthly?) may want ~2-3 months notice, even for a 6 months timelock. Doesn't mean they have to act on it, but it's nice to have imho.

TLDR: Default 20%, configurable.

**Comment by darosior** [comment]
20% sounds like a lot to me. But you do agree it needs to be relative right?

**Comment by darosior** [comment]
I know. Technically plenty of things are nice to have be configurable. I'm asking about the current state of things, without getting into further and additional work. Is it better as it is or should it be relative? Aside from saying it should be configurable it looks like you agree with me. But please speak up if not.

TL;DR: relative is not ideal, but better than what we have now. Future improvements (like making it configurable) welcome, eventually.
